backend/core/src/listings/listings.service.ts

```diff
@@ -5,13 +5,15 @@ import { Listing } from "./entities/listing.entity"
 import { ListingCreateDto, ListingUpdateDto } from "./dto/listing.dto"
 import { InjectRepository } from "@nestjs/typeorm"
 import { Repository } from "typeorm"
+import { ListingsListQueryParams } from "./listings.controller"
 
 @Injectable()
 export class ListingsService {
   constructor(@InjectRepository(Listing) private readonly repository: Repository<Listing>) {}
 
   private getQueryBuilder() {
-    return Listing.createQueryBuilder("listings")
+    return this.repository
```

Comment:
Why the switch here and elsewhere to use this.repository instead of Listing?

Reply:
`Listing` is using the entity object directly, but `this.repository` is the injected entity. That lets me swap it with a mock in the tests.